chainstate/src/detail/spend_cache/mod.rs

```diff
@@ -363,6 +414,24 @@ impl<'a, S: BlockchainStorageRead> CachedInputs<'a, S> {
 
                     // TODO: see if a different treatment should be done for different output purposes
 
+                    {
+                        let block_index = self
+                            .db_tx
+                            .get_block_index(tx_pos.block_id())
+                            .map_err(StateUpdateError::from)?
+                            .ok_or_else(|| {
+                                StateUpdateError::InvariantErrorHeaderCouldNotBeLoaded(
+                                    *tx_pos.block_id(),
+                                )
+                            })?;
+                        self.check_timelock(
```

Comment:
I think the name `verify_signatures()` is a little confusing because I wouldn't start looking for the time lock check from that function.

Reply:
I've been thinking about that for a while... I came up with the name `verify_authorization`... not sure yet. But I'm not too worried now about that. Renaming stuff is easy.